proposals/lip-0044.md

```diff
@@ -190,234 +216,320 @@ genesisDataSchema = {
+            "generatorKey": generatorKey,
+            "result": KEY_REG_RESULT_SUCCESS
+        },
+        topics=[validatorAddress]
+    )
 ```
 
 #### getValidatorAccount
 
-This function is used to retrieve information about a validator account.
-
-##### Parameters
-
-* `address`: A 20 bytes value identifying the validator account.
-
-##### Returns
-
-This function returns the validator account corresponding to the input address.
-
-##### Execution
-
-This function returns `validatorAccount(address)`. If there is no entry corresponding to `address`, it returns an empty object. If `address` is not 20 bytes long, an error is thrown.
+This function is used to retrieve information about the validator account corresponding to the input `address: Address`.
+It returns `validatorAccount(address)`. 
+If there is no entry corresponding to `address`, it returns an empty object.
 
 #### getGenesisData
 
 This function is used to retrieve information about the genesis data.
-
-##### Parameters
-
-This function has no input parameter.
-
-##### Returns
-
-This function returns the data structure stored in the genesis data substore.
+It returns the data structure stored in the genesis data substore.
 
 #### setValidatorBLSKey
 
 This function sets the BLS key of a validator account. This update is only possible if the current BLS key of the validator is equal to a predefined constant invalid BLS key.
-
-##### Parameters
-
-This function has the following input parameters in the order given below:
-
-* `validatorAddress`: A 20 bytes value identifying the validator account in the validators data substore.
-* `proofOfPossession`: The [proof of possession](https://github.com/LiskHQ/lips/blob/main/proposals/lip-0038.md#public-key-registration-and-proof-of-possession) for `blsKey`.
-* `blsKey`: A 48 bytes value, specifying the new BLS key of the validator.
-
-##### Returns
-
-* `success`: A boolean, indicating success or failure of the function execution.
-
-##### Execution
-
-This function checks that there exists a validator account registered for the input `validatorAddress`, that the input BLS key has not been registered before in the chain, that the BLS key of the validator account is equal to a certain constant invalid BLS key, and that the proof of possession for the BLS key is valid. Finally, it updates the BLS key of the validator account.
+It checks that there exists a validator account registered for the input `validatorAddress`, that the input BLS key has not been registered before in the chain, that the BLS key of the validator account is equal to a certain constant invalid BLS key, and that the input proof of possession `proofOfPossession` for the BLS key is valid. Finally, it updates the BLS key of the validator account to the input `blsKey`.
 
 ```python
-setValidatorBLSKey(validatorAddress, proofOfPossession, blsKey):
+def setValidatorBLSKey(validatorAddress: Address, proofOfPossession: ProofOfPossession, blsKey: BLSKey) -> None:
```

Comment:
Although this is not in scope for this LIP update, I find the API naming in this LIP confusing. We use `register` and `set`, where `register` is setting a new key and `set` is modifying an existing key. I think something like `add` and `update` would be much more clear.

Reply:
Moreover, in both cases we use a `Registration` event. I think it is fine to have one event for both adding / updating, but in relation to the API naming it leads to further confusion.

Reply:
we leave the terminology as is.